src/run.ts

```diff
@@ -108,6 +109,16 @@ const start = async function () {
         if (state.type === "error") console.error(`  Error: ${state.message}`);
         // Show other messages too
         if ("message" in state) console.log(`  ... ${state.message}`);
+        // PRs that need to be formatted will do so and be processed at a later time
+        if (state.type === "info" && state.needsFormatting) {
+            try {
+                await dispatchFormatAndCommit(pr.branch);
+                console.log(`  Skipping ${pr.number} because it needed to be reformatted`);
+                continue;
+            } catch (e) {
+                console.warn(`  Could not run formatting job for ${pr}: ${e}`);
+            }
+        }
```

Comment:
From https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/65993, this is the corresponding merge bot part of 2:

> Simultaneously to (2), create and merge a DefinitelyTyped pull request that:
>   1. Runs the formatter on all files in DefinitelyTyped
>   2. Adds checking in formatting fixes to the merge bot
>   3. Adds a GitHub Action that makes a new commit to format all files if needed after merges to master

Summarizing the new logic:
* https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66235/files#diff-63b72201aca662e26d65c2a8c4267dd56e12e5219c5d4dffdbabdbe6aa1ccd67 (`format-check.yml`) adds a new workflow and job on PRs that assigns the result of `dprint check` to a step summary
* `deriveStateForPR` now also returns a `needsFormatting` property for whether that new step summary indicated any unformatted file(s)
* If `needsFormatting` then this `start` logic will dispatch the other new workflow, https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66235/files#diff-76ffb347cb3582969d70785847bd57c50b9c675c6b7cb4573ac11815a31e281b (`format-and-commit.yml`), instead of taking action on the PR